src/components/DataTableSkeleton/DataTableSkeleton.js

```diff
@@ -27,6 +27,16 @@ const DataTableSkeleton = ({
     [`${prefix}--data-table-v2--compact`]: compact,
   });
 
+  let normalizedHeaders;
+
+  if (typeof headers[0] === 'object') {
+    normalizedHeaders = headers.reduce((filtered, current) =>
+      filtered.push(current.header)
```

Comment:
Thanks @emyarod! Yeah I prefer the `Array.map` solution, so I'll look into that 🕵️‍♀️ 

As for the `header` key... I didn't check for that because its existence wouldn't matter so much as the `header` itself, if that makes sense? Since I'm making a new array string with just `header` values.

Reply:
> As for the `header` key... I didn't check for that because its existence wouldn't matter so much as the `header` itself, if that makes sense? Since I'm making a new array string with just `header` values.

correct me if I'm wrong but we are trying to return an array of `header.header`s right? wouldn't it be safer to check that the `header` object is not `null` or `undefined` so that we won't run into an error when trying to access a nonexistent property of `null`/`undefined`?